features/full_tests/batch_1/internal_error_reports.feature

```diff
@@ -0,0 +1,73 @@
+Feature: Cached Error Reports
+
+Scenario: If an exception is thrown when sending errors/sessions then internal error reports should be sent
+    When I run "InternalErrorScenario"
+    And I wait to receive 1 errors
+
+    # Validate internal error report for error serialization
+    Then the error "Bugsnag-Internal-Error" header equals "true"
+
+    # Exception details
+    And the error payload field "events" is an array with 1 elements
+    And the exception "errorClass" equals "java.lang.IllegalStateException"
+    And the exception "message" equals "Mazerunner threw exception serializing error"
+    And the exception "type" equals "android"
+    And the event "unhandled" is true
+    And the event "severity" equals "error"
+    And the event "severityReason.type" equals "unhandledException"
+
+    # Internal error specific metadata
+    And the event "metaData.BugsnagDiagnostics.cacheTombstone" is false
+    And the event "metaData.BugsnagDiagnostics.filename" is not null
+    And the event "metaData.BugsnagDiagnostics.notifierName" equals "Android Bugsnag Notifier"
+    And the event "metaData.BugsnagDiagnostics.apiKey" equals "a35a2a72bd230ac0aa0f52715bbdc6aa"
+    And the event "metaData.BugsnagDiagnostics.canRead" is true
+    And the event "metaData.BugsnagDiagnostics.cacheGroup" is false
+    And the event "metaData.BugsnagDiagnostics.exists" is true
+    And the event "metaData.BugsnagDiagnostics.canWrite" is true
+    And the error payload field "events.0.metaData.BugsnagDiagnostics.usableSpace" is an integer
+    And the event "metaData.BugsnagDiagnostics.notifierVersion" is not null
+    And the event "metaData.BugsnagDiagnostics.fileLength" equals 0
+
+    # User
+    And the event "user.id" is null
+
+    # App data
+    And the event "app.buildUUID" is not null
+    And the event "app.id" equals "com.bugsnag.android.mazerunner"
+    And the event "app.releaseStage" equals "production"
+    And the event "app.type" equals "android"
+    And the event "app.version" equals "1.1.14"
+    And the event "app.versionCode" equals 34
+    And the error payload field "events.0.app.duration" is an integer
```

Comment:
Presumably we just don't have a steps for `the event {string} is an integer` (and `greater than 0` etc)?  Feel free to raise a ticket for that and we'll get them added for consistency.

Reply:
Created a ticket to make these steps.